packages/kbn-plugin-generator/sao_template/template/index.js

```diff
@@ -34,11 +39,49 @@ export default function (kibana) {
         enabled: Joi.boolean().default(true),
       }).default();
     },
-    <%_ if (generateApi) { -%>
+    <%_ if (generateApi || generateApp) { -%>
 
     init(server, options) { // eslint-disable-line no-unused-vars
+      <%_ if (generateApp) { -%>
+        const xpackMainPlugin = server.plugins.xpack_main;
+        if (xpackMainPlugin) {
+          const featureId = '<%= snakeCase(name) %>';
+
+          xpackMainPlugin.registerFeature({
+            id: featureId,
+            name: i18n.translate('<%= camelCase(name) %>.featureRegistry.featureName', {
+              defaultMessage: '<%= name %>',
+            }),
+            navLinkId: featureId,
+            icon: 'discoverApp',
```

Comment:
Is there another generic icon we can use from EUI besides the discover app?

Reply:
Works for me